TrackerApp/UI/Categories/CategoriesViewController.swift

```diff
@@ -0,0 +1,239 @@
+    var provideSelectedCategory: ((TrackerCategory) -> Void)?
+    
+    //    MARK: - LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setLayout()
+        
+        viewModel.delegate = self
+        viewModel.loadCategories()
+        
+        categoryTableView.delegate = self
+        categoryTableView.dataSource = self
+    }
+    
+    init(selectedCategory: TrackerCategory?) {
+        viewModel = CategoriesViewModel(selectedCategory: selectedCategory)
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    //   MARK: - Methods
+    private func setLayout() {
+        
+        view.backgroundColor = .white
+        
+        [titleLabel,
+         addCategoryButton,
+         categoryTableView,
+         emptyCategoriesLabel,
+         emptyCategoriesImageView
+        ].forEach {
+            view.addSubview($0)
+            $0.translatesAutoresizingMaskIntoConstraints = false
+        }
+        
+        NSLayoutConstraint.activate([
+            titleLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            titleLabel.bottomAnchor.constraint(equalTo: view.topAnchor, constant: 40),
+            
+            categoryTableView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+            categoryTableView.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 38),
+            categoryTableView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+            categoryTableView.bottomAnchor.constraint(equalTo: addCategoryButton.topAnchor, constant: -16),
+            
+            addCategoryButton.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: -50),
+            addCategoryButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 20),
+            addCategoryButton.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -20),
+            addCategoryButton.heightAnchor.constraint(equalToConstant: 60),
+            
+            emptyCategoriesImageView.centerYAnchor.constraint(equalTo: view.centerYAnchor),
+            emptyCategoriesImageView.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            
+            emptyCategoriesLabel.centerXAnchor.constraint(equalTo: view.centerXAnchor),
+            emptyCategoriesLabel.topAnchor.constraint(equalTo: emptyCategoriesImageView.bottomAnchor, constant: 8)
+        ])
+    }
+    
+    private func deleteCategory(_ category: TrackerCategory) {
+        let alert = UIAlertController(title: nil,
+                                      message: "Эта категория точно не нужна?",
+                                      preferredStyle: .actionSheet
+        )
+        let cancelAction = UIAlertAction(title: "Отмена", style: .cancel)
+        let deleteAction = UIAlertAction(title: "Удалить", style: .destructive) {[weak self] _ in
```

Comment:
давай договоримся, что у фигурной скобки всегда должны быть пробелы вокруг
```suggestion
        let deleteAction = UIAlertAction(title: "Удалить", style: .destructive) { [weak self] _ in
```